packages/web3-common/src/eth_execution_api.ts

```diff
@@ -0,0 +1,266 @@
+// The types are generated manually by referring to following doc
+// https://github.com/ethereum/execution-apis
+
+type HexString = string;
+
+// Hex encoded 32 bytes
+type HexString32Bytes = HexString;
+// Hex encoded 1 byte
+type HexStringSingleByte = HexString;
+// Hex encoded 1 byte
+type HexStringBytes = HexString;
+// Hex encoded 256 byte
+type HexString256Bytes = HexString;
+// Hex encoded unsigned integer
+type Uint = HexString;
+// Hex encoded unsigned integer 32 bytes
+type Uint256 = HexString;
+// Hex encoded address
+type Address = HexString;
+
+export interface AccessListEntry {
+	readonly [k: string]: unknown;
+	readonly address?: Address;
+	readonly storageKeys?: HexString32Bytes[];
+}
+export type AccessList = AccessListEntry[];
+export type TransactionHash = HexString;
+export type Uncles = HexString32Bytes[];
+export type BlockTag = 'earliest' | 'latest' | 'pending';
+export type BlockNumberOrTag = Uint | BlockTag;
+
+export interface BaseTransaction {
+	readonly to?: Address;
+	readonly type: HexStringSingleByte;
+	readonly nonce: Uint;
+	readonly gas: Uint;
+	readonly value: Uint;
+	readonly input: HexStringBytes;
+	readonly chainId: Uint;
+}
+
+export interface Transaction1559Unsigned extends BaseTransaction {
+	readonly maxFeePerGas: Uint;
+	readonly maxPriorityFeePerGas: Uint;
+	readonly accessList: AccessList;
+}
+
+export interface Transaction1559Signed extends Transaction1559Unsigned {
+	readonly yParity: Uint;
+	readonly r: Uint;
+	readonly s: Uint;
+}
+
+export interface Transaction2930Unsigned extends BaseTransaction {
+	readonly gasPrice: Uint;
+	readonly accessList: AccessList;
+}
+
+export interface Transaction2930Signed extends Transaction2930Unsigned {
+	readonly yParity: Uint;
+	readonly r: Uint;
+	readonly s: Uint;
+}
+
+export interface TransactionLegacyUnsigned extends BaseTransaction {
+	readonly gasPrice: Uint;
```

Comment:
Some of the calls which this type is used with (e.g. `eth_call`) `gasPrice` is optional